README
Basic AIR native extension to SteamWorks API
============================================

It's using appId 480, which is example application from the Steamworks SDK. Somehow is not possible to write to cloud with this appId, but it's working with our real one.
It's everything what we need at this moment, but there are more functions to play with in the SteamWorks SDK. I'm not sure how legal it is to include the Steam libs and headers so these are not included. You can get them at this web https://partner.steamgames.com

- - - - - - - - - - - - -
Copyright (c) 2012, Oldes
All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

CHANGELOG.md
Changelog
=========

The builds described in this file can be [downloaded here](http://dump.ventero.de/FRESteamWorks/).
A full list of implemented functions can be found by following the link in the
version number.

### [HEAD](https://github.com/Ventero/FRESteamWorks/blob/HEAD/lib/API.txt)

### [v0.7](https://github.com/Ventero/FRESteamWorks/blob/v0.7/lib/API.txt) - [Download](http://dump.ventero.de/FRESteamWorks/v0.7/)
 - Added `getSmallFriendAvatar`, `getMediumFriendAvatar` and `requestEncryptedAppTicket` - implemented by [Ilya Diev](https://github.com/kiripu) ([#43](https://github.com/Ventero/FRESteamWorks/pull/43)).
 - Added README.txt to builds describing what SDKs it was built against.
 - Don't compile SWFs in debug mode.

### [v0.6](https://github.com/Ventero/FRESteamWorks/blob/v0.6/lib/API.txt)
 - Added support for 64-bit version of Adobe AIR (20+) on Mac OS X .
 - Make code compile with VS2015 (PR #27, thanks [jonongjs](https://github.com/jonongjs)!).

### [v0.5-29-gea3356b](https://github.com/Ventero/FRESteamWorks/blob/v0.5-29-gea3356b/lib/API.txt)
 - Added `setEnv` to allow setting environment variables.
 - Fixed `enumerateUserPublishedFilesResult` to correctly return all results ([issue #13](https://github.com/Ventero/FRESteamWorks/issues/13)).
 - Implemented `indicateAchievementProgress`.
 - Removed the deprecated (and broken) `useCrashHandler`.

### [v0.5-2-gf59d5ba](https://github.com/Ventero/FRESteamWorks/blob/v0.5-2-gf59d5ba/lib/API.txt)
 - Fixed `microTxnResult` to properly return null if no results available, and correctly initialize the `authorized` property for returned objects.

### [v0.5](https://github.com/Ventero/FRESteamWorks/blob/v0.5/lib/API.txt)
 - Implemented `setOverlayNotificationPosition`
 - Added a callback for microtransaction authorizations (thanks to @Unsanctioned)
 - Added `restartAppIfNecessary` (again, thanks to @Unsanctioned)
 - Exposed `startProcess(path:String)` on Linux, which has to be called before any other API function if the API wrapper binary is in a non-standard location
 - Deprecated the (undocumented) `path` argument to `init` on Linux and `useCrashHandler`

### [v0.4-54-gdd1b399](https://github.com/Ventero/FRESteamWorks/blob/v0.4-54-gdd1b399/lib/API.txt)
 - Implemented global stats

### [v0.4-46-g8526828](https://github.com/Ventero/FRESteamWorks/blob/v0.4-46-g8526828/lib/API.txt)
 - Fixed `getStat{Int,Float}` on Linux (which would previously always return 0)
 - Added support for basic authentication & ownership checks

### [v0.4-36-g5ff1e97](https://github.com/Ventero/FRESteamWorks/blob/v0.4-36-g5ff1e97/lib/API.txt)
 - Added `findOrCreateLeaderboard`
 - Implemented basic friends API

### [v0.4-30-g207935d](https://github.com/Ventero/FRESteamWorks/blob/v0.4-30-g207935d/lib/API.txt)
 - Handle certain cases of API failures more gracefully
 - Implemented `getEnv`

### [v0.4-19-g0cc78a0](https://github.com/Ventero/FRESteamWorks/blob/v0.4-19-g0cc78a0/lib/API.txt)
 - Work around a bug in Flash Builder where certain classes from the ANE would be ignored.

### [v0.4](https://github.com/Ventero/FRESteamWorks/blob/v0.4/lib/API.txt)
 - Added Steam Workshop support
 - Various leaderboards related functions added
 - Added support for some DLC related functions
 - Implemented several `activateGameOverlayTo___` functions


src/functions.h
/* automatically generated by generateAPI.rb */
/* START GENERATED CODE */
/********************************/
/* General Steamworks functions */
/********************************/
X(AIRSteam_Init) /* = 0 */
X(AIRSteam_RunCallbacks) /* = 1 */
X(AIRSteam_GetUserID) /* = 2 */
X(AIRSteam_GetAppID) /* = 3 */
X(AIRSteam_GetAvailableGameLanguages) /* = 4 */
X(AIRSteam_GetCurrentGameLanguage) /* = 5 */
X(AIRSteam_GetPersonaName) /* = 6 */
X(AIRSteam_RestartAppIfNecessary) /* = 7 */
/************************/
/* Stats / Achievements */
/************************/
X(AIRSteam_RequestStats) /* = 8 */
X(AIRSteam_SetAchievement) /* = 9 */
X(AIRSteam_ClearAchievement) /* = 10 */
X(AIRSteam_IsAchievement) /* = 11 */
X(AIRSteam_IndicateAchievementProgress) /* = 12 */
X(AIRSteam_GetStatInt) /* = 13 */
X(AIRSteam_GetStatFloat) /* = 14 */
X(AIRSteam_SetStatInt) /* = 15 */
X(AIRSteam_SetStatFloat) /* = 16 */
X(AIRSteam_StoreStats) /* = 17 */
X(AIRSteam_ResetAllStats) /* = 18 */
X(AIRSteam_RequestGlobalStats) /* = 19 */
X(AIRSteam_GetGlobalStatInt) /* = 20 */
X(AIRSteam_GetGlobalStatFloat) /* = 21 */
X(AIRSteam_GetGlobalStatHistoryInt) /* = 22 */
X(AIRSteam_GetGlobalStatHistoryFloat) /* = 23 */
/****************/
/* Leaderboards */
/****************/
X(AIRSteam_FindLeaderboard) /* = 24 */
X(AIRSteam_FindOrCreateLeaderboard) /* = 25 */
X(AIRSteam_FindLeaderboardResult) /* = 26 */
X(AIRSteam_GetLeaderboardName) /* = 27 */
X(AIRSteam_GetLeaderboardEntryCount) /* = 28 */
X(AIRSteam_GetLeaderboardSortMethod) /* = 29 */
X(AIRSteam_GetLeaderboardDisplayType) /* = 30 */
X(AIRSteam_UploadLeaderboardScore) /* = 31 */
X(AIRSteam_UploadLeaderboardScoreResult) /* = 32 */
X(AIRSteam_DownloadLeaderboardEntries) /* = 33 */
X(AIRSteam_DownloadLeaderboardEntriesResult) /* = 34 */
/**************************/
/* Cloud / Remote Storage */
/**************************/
X(AIRSteam_GetFileCount) /* = 35 */
X(AIRSteam_GetFileSize) /* = 36 */
X(AIRSteam_FileExists) /* = 37 */
X(AIRSteam_FileWrite) /* = 38 */
X(AIRSteam_FileRead) /* = 39 */
X(AIRSteam_FileDelete) /* = 40 */
X(AIRSteam_FileShare) /* = 41 */
X(AIRSteam_FileShareResult) /* = 42 */
X(AIRSteam_IsCloudEnabledForApp) /* = 43 */
X(AIRSteam_SetCloudEnabledForApp) /* = 44 */
X(AIRSteam_GetQuota) /* = 45 */
/******************/
/* UGC / Workshop */
/******************/
X(AIRSteam_UGCDownload) /* = 46 */
X(AIRSteam_UGCRead) /* = 47 */
X(AIRSteam_GetUGCDownloadProgress) /* = 48 */
X(AIRSteam_GetUGCDownloadResult) /* = 49 */
X(AIRSteam_PublishWorkshopFile) /* = 50 */
X(AIRSteam_PublishWorkshopFileResult) /* = 51 */
X(AIRSteam_DeletePublishedFile) /* = 52 */
X(AIRSteam_GetPublishedFileDetails) /* = 53 */
X(AIRSteam_GetPublishedFileDetailsResult) /* = 54 */
X(AIRSteam_EnumerateUserPublishedFiles) /* = 55 */
X(AIRSteam_EnumerateUserPublishedFilesResult) /* = 56 */
X(AIRSteam_EnumeratePublishedWorkshopFiles) /* = 57 */
X(AIRSteam_EnumeratePublishedWorkshopFilesResult) /* = 58 */
X(AIRSteam_EnumerateUserSubscribedFiles) /* = 59 */
X(AIRSteam_EnumerateUserSubscribedFilesResult) /* = 60 */
X(AIRSteam_EnumerateUserSharedWorkshopFiles) /* = 61 */
X(AIRSteam_EnumerateUserSharedWorkshopFilesResult) /* = 62 */
X(AIRSteam_EnumeratePublishedFilesByUserAction) /* = 63 */
X(AIRSteam_EnumeratePublishedFilesByUserActionResult) /* = 64 */
X(AIRSteam_SubscribePublishedFile) /* = 65 */
X(AIRSteam_UnsubscribePublishedFile) /* = 66 */
X(AIRSteam_CreatePublishedFileUpdateRequest) /* = 67 */
X(AIRSteam_UpdatePublishedFileFile) /* = 68 */
X(AIRSteam_UpdatePublishedFilePreviewFile) /* = 69 */
X(AIRSteam_UpdatePublishedFileTitle) /* = 70 */
X(AIRSteam_UpdatePublishedFileDescription) /* = 71 */
X(AIRSteam_UpdatePublishedFileSetChangeDescription) /* = 72 */
X(AIRSteam_UpdatePublishedFileVisibility) /* = 73 */
X(AIRSteam_UpdatePublishedFileTags) /* = 74 */
X(AIRSteam_CommitPublishedFileUpdate) /* = 75 */
X(AIRSteam_GetPublishedItemVoteDetails) /* = 76 */
X(AIRSteam_GetPublishedItemVoteDetailsResult) /* = 77 */
X(AIRSteam_GetUserPublishedItemVoteDetails) /* = 78 */
X(AIRSteam_GetUserPublishedItemVoteDetailsResult) /* = 79 */
X(AIRSteam_UpdateUserPublishedItemVote) /* = 80 */
X(AIRSteam_SetUserPublishedFileAction) /* = 81 */
/***********/
/* Friends */
/***********/
X(AIRSteam_GetFriendCount) /* = 82 */
X(AIRSteam_GetFriendByIndex) /* = 83 */
X(AIRSteam_GetFriendPersonaName) /* = 84 */
X(AIRSteam_GetSmallFriendAvatar) /* = 85 */
X(AIRSteam_GetMediumFriendAvatar) /* = 86 */
/******************************/
/* Authentication & Ownership */
/******************************/
X(AIRSteam_GetAuthSessionTicket) /* = 87 */
X(AIRSteam_GetAuthSessionTicketResult) /* = 88 */
X(AIRSteam_BeginAuthSession) /* = 89 */
X(AIRSteam_EndAuthSession) /* = 90 */
X(AIRSteam_CancelAuthTicket) /* = 91 */
X(AIRSteam_UserHasLicenseForApp) /* = 92 */
X(AIRSteam_RequestEncryptedAppTicket) /* = 93 */
X(AIRSteam_GetEncryptedAppTicket) /* = 94 */
/***********/
/* Overlay */
/***********/
X(AIRSteam_ActivateGameOverlay) /* = 95 */
X(AIRSteam_ActivateGameOverlayToUser) /* = 96 */
X(AIRSteam_ActivateGameOverlayToWebPage) /* = 97 */
X(AIRSteam_ActivateGameOverlayToStore) /* = 98 */
X(AIRSteam_ActivateGameOverlayInviteDialog) /* = 99 */
X(AIRSteam_IsOverlayEnabled) /* = 100 */
X(AIRSteam_SetOverlayNotificationPosition) /* = 101 */
/***********************/
/* DLC / subscriptions */
/***********************/
X(AIRSteam_IsSubscribedApp) /* = 102 */
X(AIRSteam_IsDLCInstalled) /* = 103 */
X(AIRSteam_GetDLCCount) /* = 104 */
X(AIRSteam_InstallDLC) /* = 105 */
X(AIRSteam_UninstallDLC) /* = 106 */
X(AIRSteam_DLCInstalledResult) /* = 107 */
/********************/
/* Microtransaction */
/********************/
X(AIRSteam_MicroTxnResult) /* = 108 */
/**********************************/
/* Other non-Steamworks functions */
/**********************************/
X(AIRSteam_GetEnv) /* = 109 */
X(AIRSteam_SetEnv) /* = 110 */
/* END GENERATED CODE */
